server/py/framework/db/sqldb/db.py

```diff
@@ -5112,21 +5104,73 @@ def _get_class_instance_by_uid(self, session, cls, name, project, uid):
         return query.one_or_none()
 
     def _get_mep_latest_instance(
-        self, session, cls, name, function_name, project, function_tag
+        self, session, cls, name, function_name, project, function_tag, _get_query=False
     ):
         query = (
             session.query(cls)
-            .join(cls.Tag)
-            .filter(
-                cls.project == project,
-                cls.name == name,
-                cls.function_name == function_name,
-                cls.function_tag == function_tag,
-                cls.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+            .options(
+                joinedload(cls.function),
+                joinedload(cls.model),
+                joinedload(cls.tags),
             )
+            .filter(cls.project == project, cls.name == name)
         )
 
-        return query.one_or_none()
+        # Apply function name filter (must join Function first)
+        if function_name:
+            query = query.join(Function).filter(Function.name == function_name)
+
+        # Apply function tag filter (must join Function.tags first)
+        if function_tag:
+            query = query.join(Function.tags).filter(Function.Tag.name == function_tag)
+
+        # Apply latest tag filter
+        query = query.join(cls.tags).filter(
+            cls.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST
+        )
+
+        if _get_query:
+            return query
+
+        return query.first()  # Use `.first()` instead of `.one_or_none()` for safety
+
+    def _get_mep_instances(
+        self, session, cls, name, function_name, project, function_tag, _get_query=False
```

Comment:
both `function_name` and `function_tag` are opt. Update the docstrings order as well. 
```suggestion
        self, session, cls, name,  project, function_name=None, function_tag=None, _get_query=False
```

Reply:
you didn't fix it..